packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -268,9 +268,23 @@ const App = (props) => {
         const path = buildUrl('/account/wishlist')
         history.push(path)
     }
+
+    // START - Demo "onContextChange" handler.
+    const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms))
+    const contextChangeHandler = async (context) => {
+        const waitFor = 7500 // ms
+        console.log(
+            `STOREFRONT: Handling context change. Synthetic ${waitFor / 1000}s wait.`,
+            context
+        )
+        await sleep(waitFor)
+        console.log('STOREFRONT: Finished custom context handler.')
+    }
+    // END
+
```

Comment:
This will be removed before merging, but this is simply mimicking a network request (that takes a long time apparently)

Reply:
Before going ahead the PR, Can we extend the demo by making a real request to the CMS and see the content changed? It could help us identify any missing actions we typically need to take to set the context in a CMS using a real API.
I'm thinking about scenarios where the request returns a 400 or 500 response and what we want to do in those states.

Reply:
So the interesting part is that we don't really care about network responses failing per-say because our API is generically async.. yes a network request is async, but there are a lot of other things that at async too. 

It's possible for example that someone implements their `onContextHandler` in a way that catches their network errors thus bypassing our built in error handling. 

But I've updated the demo code to be more network like you can play with it [here](https://runtime-admin-preview.mobify-storefront.com/kyle-preview-demo/p/test-ben/preview). In my change the context handler makes a "fake" network request, there are 3 outcomes after clicking the preview button:

1. Everything goes as planned an we see the page refresh.
2. The fake request takes longer than our built in 10 second timeout. In this case we should see an error dialog that says we expected a message response but didn't get one.
3. The fake request throws an error, this emulates a 404, 500, whatever. Here we should see the error dialog pop up with the error message of "A random network error occured!" (with the spelling mistake included lol)